go/rpc/encrypted_client.go

```diff
@@ -147,45 +147,53 @@ func (c *EncRPCClient) Subscribe(ctx context.Context, namespace string, ch inter
 		return nil, fmt.Errorf("expected a channel of type `chan types.Log`, got %T", ch)
 	}
 	clientChannel := make(chan common.IDAndEncLog)
-	subscription, err := c.obscuroClient.Subscribe(ctx, namespace, clientChannel, subscriptionType, encryptedParams)
+	subscription, err := c.obscuroClient.Subscribe(ctx, nil, namespace, clientChannel, subscriptionType, encryptedParams)
 	if err != nil {
 		return nil, err
 	}
 
-	go func() {
-		for {
-			select {
-			case idAndEncLog := <-clientChannel:
-				jsonLogs, err := c.decryptResponse(idAndEncLog.EncLog)
-				if err != nil {
-					log.Error("could not decrypt logs received from subscription. Cause: %s", err)
-					continue
-				}
+	err = c.setResultToSubID(clientChannel, result, subscription)
+	if err != nil {
+		subscription.Unsubscribe()
+		return nil, err
+	}
 
-				var logs []*types.Log
-				err = json.Unmarshal(jsonLogs, &logs)
-				if err != nil {
-					log.Error("could not unmarshal log from `data` field of log received from subscription. "+
-						"Data field contents: %s. Cause: %s", string(jsonLogs), err)
-					continue
-				}
+	go c.forwardLogs(clientChannel, logCh, subscription)
 
-				for _, decryptedLog := range logs {
-					idAndLog := common.IDAndLog{
-						SubID: idAndEncLog.SubID,
-						Log:   decryptedLog,
-					}
-					logCh <- idAndLog
-				}
+	return subscription, nil
+}
 
-			case <-subscription.Err(): // This channel's sole purpose is to be closed when the subscription is unsubscribed.
-				log.Error("subscription closed")
-				break
+func (c *EncRPCClient) forwardLogs(clientChannel chan common.IDAndEncLog, logCh chan common.IDAndLog, subscription *rpc.ClientSubscription) {
```

Comment:
All the logic in `forwardsLogs` is unchanged, the diff just comes up a bit ugly.